### ISSUE TITLE
Unbalanced calls to begin/end appearance transitions for -

### ISSUE THREAD

**Issue by nExmond** [jonkykong/SideMenu#373]
## New Issue Checklist

I have read the [guidelines for contributing](https://github.com/jonkykong/SideMenu/blob/master/.github/CONTRIBUTING.md) and I understand:
- [x] My issue is happening in the **latest version** of SideMenu.
- [x] My issue was **not** solved in the [README](https://github.com/jonkykong/SideMenu/blob/master/README.md).
- [x] My issue can **not** be answered on [stackoverflow.com](stackoverflow.com).
- [x] My issue is **not** a request for new functionality that I am unwilling to build and contribute with a pull request.
- [x] My issue **is** reproducible in the [demo project](https://github.com/jonkykong/SideMenu/tree/master/Example).

## Issue Description

**How to reproduce**
Place the view with auto-layout on the view controller associated with the first cell of UITableViewController in the Demo.
When you activate the side menu by Menu Button  and immediately select the first cell, the view controller is displayed normally.
However, when you select a cell after a short swipe, the layout is broken and the following warning message is printed to the console.
**Unbalanced calls to begin/end appearance transitions for <Example.MainViewController: 0x105d05470>.**

Attached a gif.

![gif](https://images2.imgbox.com/61/2a/hX1kE1Tp_o.gif)

**Comment by jonkykong** [comment]
Hi @nExmond,

Thanks for reporting this. Unfortunately, I'm not able to reproduce this problem with the instructions you've provided. This is also the first report of such an issue. When I try to repeat those steps everything appears normally with no warnings.

Can you tell me what I'm missing with these steps?

Prerequisite: I added a view to the first view controller (red). I set its auto-resize mask and didn't use constraints.

1. I tap "Left Menu".
2. I select the first cell in the SideMenu. The red screen appears as expected.
3. I hit back.
4. I tap "Left Menu".
5. I drag the main screen back over the SideMenu, but then cancel and leave it open.
6. I select the first cell in the SideMenu. The red screen appears as expected.

This was on simulator using iOS 11.4.

**Comment by nExmond** [comment]
You can check the auto-layout by applying the following.

![auto layout](https://images2.imgbox.com/4f/19/yVBQCrHM_o.png)

Perhaps due to a transition error, the Navigation Bar appears to be late and the top space is adjusted.

The strange thing is that if you activate the menu with a swipe gesture and then reproduce the error, there is nothing wrong with it.

**Comment by jonkykong** [comment]
Okay, I've been able to reproduce the problem when constraints are used, specifically when the top constraint is anchored to the Safe Area. If it's anchored to the superview there isn't an issue.

Off the top of my head, I don't have any explanation for this and it seems highly likely that this is a transition bug on Apple's part. The library behaves correctly most of the time, but doesn't if the gesture is used and aborted. This is where `cancel()` is called, aborting the transition, returning the menu back to its displayed state that it was at before the gesture was used. There's no modification to the navigation or layout. The main screen continues to display correctly despite the layout being off on the transitioning-to screen.

Unfortunately I don't have the cycles to investigate this or come up with a fix. This is a corner case aesthetic bug, though by my own standards I'd definitely like to see it fixed. If you'd like to investigate further to potentially find a root cause and/or submit a PR, it would be extremely helpful.

Minor note: I'm not getting any warnings about transitions being unbalanced.

**Comment by nExmond** [comment]
I have tested multiple MacBooks and have always had an unbalanced alert. But it is amazing that you do not appear.
Fixed the problem of changing the vertical position by restricting the superview as instructed.

I have to worry about solving the fundamental problem.
thanks.

**Comment by jonkykong** [comment]
This issue should now be resolved in version 6.0.0+ of SideMenu. Please run `pod update` and reopen this issue if the problem persists.
